### DISCUSSION TITLE
message reaction could not get user's selected reaction from cache  after rebooting bot

### DISCUSSION THREAD

**Discussion by lokyse** [discordjs/discord.js#8844] [Q&A]
### Which package is this bug report for?

discord.js

### Issue description

User  choose some reaction in the message,  so the reaction  records  the user's info, and we could use ` messageReaction.users.remove(user.id)` method to  remove this user(who has selected the reaction) by id in the reaction that mean user unselect the reaction.  
 
The issue it that   if I reboot bot program ,   The program could fetch  message and  reactions belong to  message at 'messageReactionAdd' event handler. But  the reaction don't have those user who has selected it before reboot bot program. 
I see ` MessageReaction.users.cache.get(user.id)` method   don't  give me the  user  as expected  instead of 'undefine' value. 

### Code sample

```typescript
client.on("messageReactionAdd", async (reaction, user) => {
            if (user.id === client.user.id) return;

            if (reaction.partial) {
                  // If the message this reaction belongs to was removed, the fetching might result in an API error which should be handled
                  try {
                        await reaction.fetch();
                  } catch (error) {
                        console.error(
                              "Something went wrong when fetching the message:",
                              error
                        );
                        return;
                  }
            }

//   omit  code ....
// ....



  const userReactions =
                                          reaction.message.reactions.cache.filter(
                                                (reaction) =>
                                                      reaction.emoji.name ==
                                                            selected.choiceEmoji &&
                                                      reaction.message.id ==
                                                            selected.pollId
                                          );


                                    userReactions.forEach((e) => {
                                          // has value
                                          console.log(e.emoji.name, e.count);
        
                                          // not work here!  got 'undefine' 
                                          e.users.cache.get(user.id)
 
                                         // e.users.remove(user.id);
                                    });
```

### Package version

^14.6.0"

### Node.js version

v18.12.0

### Operating system

macos catalina  10.15.7

### Priority this issue should have

High (immediate attention needed)

### Which partials do you have configured?

User, Channel, Message, Reaction

### Which gateway intents are you subscribing to?

Guilds, GuildEmojisAndStickers, GuildMessages, GuildMessageReactions, DirectMessages, DirectMessageReactions, MessageContent

**Comment by Jiralite** [comment]
Seems like you don't know this exists: [`ReactionUserManager#fetch()`](https://discord.js.org/#/docs/discord.js/14.6.0/class/ReactionUserManager?scrollTo=fetch)

**Reply by lokyse** [reply]
> Seems like you don't know this exists: [`ReactionUserManager#fetch()`](https://discord.js.org/#/docs/discord.js/14.6.0/class/ReactionUserManager?scrollTo=fetch)


```
client.on("messageReactionAdd", async (reaction, user) => {
            if (user.id === client.user.id) return;
            // I add user.fetch here,   right?
            await reaction.users.fetch();
            if (reaction.partial) {
```
But It still not work,   'e.users.cache.get(user.id)' give me undefine

**Comment by ga1aXy-pog** [comment]
I would say that ReactionUserManager#fetch() system through the current version of discord.js should work.
Check discord.js logs for this question for better approval and understanding.
